tests/EStimLibrary.UnitTests/Extensions/SpatialModel/StringHierarchy/StringHierarchyAreaFactoryTests.cs

```diff
@@ -0,0 +1,268 @@
+﻿using EStimLibrary.Core.SpatialModel;
+using EStimLibrary.Extensions.SpatialModel.StringHierarchy;
+
+namespace EStimLibrary.UnitTests.Extensions.SpatialModel.StringHierarchy;
+
+/// <summary>
+/// Test class for StringHierarchyAreaFactory.
+/// </summary>
+public class StringHierarchyAreaFactoryTests
+{
+    private readonly ITestOutputHelper _output;
+
+    // Test class constructor creates an output helper so can write console
+    // output.
+    public StringHierarchyAreaFactoryTests(
+        ITestOutputHelper testOutputHelper)
+    {
+        this._output = testOutputHelper;
+    }
+
+    // Test method naming convention: LibClassMethodName_ScenarioShouldExpectn
+
+    /// <summary>
+    /// Test the constructor with null parameter.
+    /// </summary>
+    [Fact]
+    public void Constructor_ShouldAcceptNull()
```

Comment:
We actually want to disallow null input. Is the current implementation (pre your related changes in the source code files above) allowing it?